File: 205-isomorphic-strings/205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        
        unordered_map<char, char> mp;
        unordered_map<char, bool> used;

        for(int i=0; i<s.length(); i++){

            if(mp.find(s[i]) != mp.end()){

                if(mp[s[i]] != t[i]){
                    return false;
                }
            }
            else{
                if(used.find(t[i]) != used.end()){
                    return false;
                }
                else{
                    mp[s[i]] = t[i];
                    used[t[i]] = true;
                }
            }
        }

        return true;
    }
};

File: 205-isomorphic-strings/205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        
        Map<Character, Character> map = new HashMap<>();
        Map<Character, Boolean> used = new HashMap<>();
        
        for(int i=0; i<s.length(); i++){
            
            char sch = s.charAt(i);
            char tch = t.charAt(i);
            
            if(map.containsKey(sch) == true){
                
                if(map.get(sch) != tch) return false;
            }
            else{
                if(used.containsKey(tch) == true){
                    return false;
                }
                else{
                    map.put(sch, tch);
                    used.put(tch, true);
                }
            }
        }
        
        return true;
    }
}